src/calculate-totals.js

```diff
@@ -0,0 +1,148 @@
+const calculateTotals = (result) => {
+  if (result.data.length === 0) {
+    return result
+  }
+
+  let totals = {}
+
+  // Sum up simple columns
+  if ("users" in result.data[0]) {
+    totals.users = _sumColumn({ column: "users", result })
+  }
+  if ("visits" in result.data[0]) {
+    totals.visits = _sumColumn({ column: "visits", result })
+  }
+
+  // Sum up categories
+  if (result.name.match(/^device_model/)) {
+    totals.device_models = _sumVisitsByColumn({
+      column: "mobile_device",
+      result,
+    })
+  }
+  if (result.name.match(/^language/)) {
```

Comment:
Originally this used lodash. I'm wondering if we need to check that the reports name starts with a given substring, or if these can be strict comparisons?

I don't see a report that matches the substring prefix without matching the entire string?